astromodels/functions/functions.py

```diff
@@ -1367,3 +1371,313 @@ def _set_units(self, x_unit, y_unit):
 
     def evaluate(self, x, K, xc):
         return K * np.exp(np.divide(x, -xc))
+
+class DMFitFunction(Function1D):
+    r"""
+        description :
+        
+            Class that evaluates the spectrum for a DM particle of a given
+            mass, channel, cross section, and J-factor. Based on standard
+            Fermi Science Tools function DMFitFunction. Note input table only
+            calculated spectra up to m_DM of 10 TeV
+            
+            The parameterization is given by
+            
+            F(x) = 1 / (8 * pi) * (1/mass^2) * sigmav * J * dN/dE(E,mass,i)
+            
+            Note that this class assumes that mass and J-factor are provided
+            in units of GeV and GeV^2 cm^-5
+            
+        latex : $$
+        
+        parameters :
+        
+            mass :
+                desc : DM mass (GeV)
+                initial value : 10
+                fix : yes
+                
+            channel :
+                desc : DM annihilation channel
+                initial value : 4
+                fix : yes
+            
+            sigmav : 
+                desc : DM annihilation cross section (cm^3/s)
+                initial value : 1.e-26
+            
+            J :
+                desc : Target total J-factor (GeV^2 cm^-5)
+                initial value : 1.e20
+                fix : yes
+        """
+    
+    __metaclass__ = FunctionMeta
+    
+    def _setup(self):
+        
+        astroDir = astromodels.__file__
```

Comment:
There are more robust ways of dealing with data files. This one is not passing the automatic testing, because the data files is not found on the testing machine (TrevisCI) for example.

However, I am going to fix it myself on the trunk so we speed up the iteration.